### PR TITLE
Short hand notation for setters/subst

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent 

This PR is to introduce new short hand notation for setters/substitutions in resource files. The scope of this PR is for cfg set operation where we look for short hand ref and if it is not present we fall back to the current openAPI ref format. Once we agree upon the notation I can have PRs for list, create-setter/subst commands.

https://github.com/GoogleContainerTools/kpt/issues/543

### CODE REVIEW

File: kyaml/fieldmeta/fieldmeta.go [hunk 1]
```diff
@@ -166,3 +217,20 @@ func (it FieldValueType) TagForValue(value string) string {
...
+	SetterDefinitionPrefix = CLIDefinitionsPrefix + "setters."
+
+	// SubstitutionDefinitionPrefix is the prefix for substitution definition keys.
+	SubstitutionDefinitionPrefix = CLIDefinitionsPrefix + "substitutions."
+
+	// DefinitionsPrefix is the prefix used to reference definitions in the OpenAPI
+	DefinitionsPrefix = "#/definitions/"
+
+	// ShortHandRef is the shorthand reference to setters and substitutions
+	ShortHandRef = "$openAPI"
```
Comment (phanimarupaka) [inline kyaml/fieldmeta/fieldmeta.go:235]: We need to decide on the ShortHandRef name

Comment (frankfarzan) [reply kyaml/fieldmeta/fieldmeta.go:236]: How about `kpt-set` to tie this back into the concept of a `set` in KPT?

Comment (phanimarupaka) [reply kyaml/fieldmeta/fieldmeta.go]: @frankfarzan Updated PR with getter and setter for ShortHandRef. However, this ref should be set in upstream kpt repo as there shouldn't be any kpt references in kustomize. Will make that change once this is merged.

File: kyaml/fieldmeta/fieldmeta.go [hunk 2]
```diff
@@ -76,6 +81,53 @@ func (fm *FieldMeta) Read(n *yaml.RNode) error {
 	return nil
 }
 
+// processShortHand parses the comment for short hand ref and loads schema to fm
+func (fm *FieldMeta) processShortHand(comment string) bool {
```
Comment (mortent) [inline kyaml/fieldmeta/fieldmeta.go:85]: We should document the meaning of the return value.

File: kyaml/fieldmeta/fieldmeta.go [hunk 3]
```diff
@@ -76,6 +81,53 @@ func (fm *FieldMeta) Read(n *yaml.RNode) error {
...
+	if err != nil {
+		return false
+	}
+	name := input[shortHandRef]
+	if name == "" {
+		return false
+	}
+
+	// check if setter with the name exists, else check for a substitution
+	// setter and substitution can't have same name in shorthand
```
Comment (mortent) [inline kyaml/fieldmeta/fieldmeta.go:97]: Don't we already prohibit setters and substitutions with the same name?

File: cmd/config/internal/commands/cmdset_test.go [hunk 4]
```diff
@@ -129,10 +129,10 @@ apiVersion: v1alpha1
 kind: Example
 openAPI:
   definitions:
-    io.k8s.cli.setters.image:
+    io.k8s.cli.setters.image-setter:
```
Comment (mortent) [inline cmd/config/internal/commands/cmdset_test.go:132]: This change doesn't influence the behavior right? So we are not introducing a requirement that setters must have a `-setter` suffix.

File: kyaml/fieldmeta/fieldmeta.go (hunk 2 above)
Comment (mortent) [reply kyaml/fieldmeta/fieldmeta.go:85]: Is it always safe to just return false whenever we encounter an error?

File: kyaml/fieldmeta/fieldmeta.go (hunk 3 above)
Comment (phanimarupaka) [reply kyaml/fieldmeta/fieldmeta.go:97]: Yes. So only we have to do this check as we don't know if it is setter or substitution. So we check if it is setter or substitution.

File: cmd/config/internal/commands/cmdset_test.go (hunk 4 above)
Comment (phanimarupaka) [reply cmd/config/internal/commands/cmdset_test.go:132]: No. It is just a name change for tests readability.

File: kyaml/fieldmeta/fieldmeta.go (hunk 2 above)
Comment (phanimarupaka) [reply kyaml/fieldmeta/fieldmeta.go:87]: Yes. We should not throw error in any case as the comment might not be related to kpt/setters. We do the best to parse it and see if it is setter ref and continue if it is not.

Comment (phanimarupaka) [reply kyaml/fieldmeta/fieldmeta.go:87]: Updated the comment.

**Pull request merged**